engines/bastion_katello/app/assets/javascripts/bastion_katello/products/details/repositories/required-tags.service.js

```diff
@@ -0,0 +1,59 @@
+
+/**
+ * @ngdoc service
+ * @name  Bastion.products.details.repositories.service:RequiredTags
+ *
+ * @description
+ *   Helper functions for repo requiredTags.
+ */
+
+angular
+    .module('Bastion.repositories')
+    .service('RequiredTags', function () {
+
+        this.isRequiredTagSelected = function (tag, repo) {
+            if (!repo.required_tags) {
+                return false;
+            }
+            return !!repo.required_tags.find(function (reqTag) {
+                return reqTag === tag;
+            });
+        };
+
+        this.getRequiredTagsOptions = function () {
```

Comment:
What happens when someone uses the API to set a value that isn't represented here?

Reply:
hmm..

I think it would break things.  It wouldn't be shown as an option, and then would be overwritten when you saved the repo in the UI.

Reply:
@jturel good catch :)

Reply:
@jturel Updated to address this.  If a repo has any tags in addition to the defaults, `getRequiredTagsOptions` will now honor them and it should otherwise work like normal.